install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml

```diff
@@ -278,13 +278,17 @@ spec:
               - NET_RAW
               # Used since cilium monitor uses mmap
               - IPC_LOCK
+              {{- if .Values.securityContext.capsysmodule }}
               # Used in iptables. Consider removing once we are iptables-free
               - SYS_MODULE
```

Comment:
I wonder if it's possible to express this list of capabilities in helm in some sort of array/slice structure, I'm thinking in the helm values ideally something like:

```
securityContext:
  capabilities:
    - CHOWN
    - KILL
    - NET_ADMIN
    - SYS_MODULE
    - ...
```

Then (a) if the user doesn't specify the list, it populates with some default value like the current list. And (b) in the Talos case you could override with the exact permissions in the values file.

This might help also in future for stuff like PERFMON, BPF capabilities which are commented out below but are only available with newer kernel + container runtimes.

Food for thought, otherwise we may need to come back and add more and more specific booleans for each capability into the helm charts.

Reply:
If this makes sense and is easy to do, that would be good. Otherwise if it's not a good idea, the current PR looks fine :+1:

Reply:
what do you think about moving SYS_MODULE from the default capability list to Values.securityContext.extraCapabilities?

AFAIK, the default list is a minimum list of capabilities for cilium to start.

Reply:
@joestringer : totally agree, I think moving all the caps to `.securityContext.capabilities` makes a lot of sense and it's easy to do. Don't know why I haven't thought of that in the first place ;-)

I'll update the PR with that suggestion.

Reply:
@sayboras SYS_MODULE is needed for some iptables functionality, I think it will cause Cilium to fail pretty quickly if it is not present.